content/asp.net/getting-started/results.md

```diff
@@ -1,20 +1,88 @@
+
+```c#
+[Route("api/person/{id?}")]
+public class PersonApiController : Controller
+{
+    // ...
+}
+``` 
+
+The class has a route on it that has `api` at its base URL and accepts an optional `id` parameter ([more about routing](routing.md)). The framework takes away some of work required to serialize your data that you wish to return, allowing you to do something like the following and returning a simple type to your client:
+
+```c#
+public Person Get(int id)
+{
+    // load person, usually from a service
+    var person = new Person { Id = id, FirstName = "Johhny", LastName = "B. Goode." };
+    return person;                      
+}
+```
+
+You can try that out in your browser (navigate to `/api/person` ) and you will see the result; however, this is a simple action that creates an object and returns it directly. There is very little chance that something would go wrong, but in a more typical scenario we'd be talking to a service, doing validation on the incoming parameters and such. What happens in the above code if the person is not found in the service? Or if there is an error connecting to the database? If the request doesn't contain a valid ID?
+
+By returning an object like above the framework will kindly serialize our object, return it in JSON format and set the status code to HTTP 200. Should an error occur, an HTTP 500 will be returned. Those are our only options. It is best, then, to leverage the framework and create responses that are more suitable for use by our client.
+
+```c#
+public IActionResult Get(int id)
+{
+    // load person, usually from a service
+    var person = new Person { Id = id, FirstName = "Johhny", LastName = "B. Goode." };
+    return Ok(person);
+}
+```
+
+There are only two small changes in the above code, the function's return type and wrapping `person` with the call to `Ok()`, but we've added a lot of flexibility. Let's consider a more complex scenario where a call to find `person` to a service doesn't yield a result.
+
+```c#
+public IActionResult Get(int id)
+{
+    if (id <= 0)
+        return BadRequest("You must specify an ID > 0.");
+
+    var person = _personService.GetById(id);
+    if (person == null)
+        return NotFound();
+
+    return Ok(person);
+}
+```
+
+Now, by using `IActionResult`, we have the flexibility to introduce alternate paths to respond to the request. We use the helper methods of the base `Controller` class to generate results based on the request. In the event that an invalid ID is presented, we can return an HTTP 400 signalling the client made an error. When we can't locate the record via the service, we return an HTTP 404 saying that the resource cannot be found. Finally, the HTTP 200 status code is explicitly set with with `Ok(person)` invocation. 
+
+You can read in more detail about the use of `IActionResult` in the context of Web API on the [ASP.NET Docs Site](https://docs.asp.net/en/latest/mvc/models/formatting.html).  
```

Comment:
You can also learn how to make these policies explicit and encapsulate them in filters in [this MSDN article] (https://msdn.microsoft.com/en-us/magazine/mt767699.aspx?f=255&MSPPError=-2147217396).